src/adapters/store.js

```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
-    throw e
+  async checkInit(target, name, descriptor) {
+    if (this.loggedIn && !this.masqInitialized) {
+      const alreadyLoggedIntoMasq = await this.masqStore.isLoggedIn()
+      if (alreadyLoggedIntoMasq) {
+        this.abstractStore = this.masqStore
+      }
+      await this.masqInitPromise
+      this.masqInitialized = true
+    }
   }
-}
 
-Store.prototype.getLastLocation = async function() {
-  try {
-    return await abstractStore.get(`qmaps_v${version}_last_location`)
-  } catch (e) {
-    Error.sendOnce('store', 'getLastLocation', 'error getting location', e)
-    return null
+  async login() {
+    await this.checkInit()
+    try {
+      const loginParams = {
+        endpoint: this.masqConfig.endpoint,
+        url: window.location.origin + window.location.pathname,
+        title: this.masqConfig.title,
+        desc: this.masqConfig.desc,
+        icon: this.masqConfig.icon
+      }
+
+      await this.masqStore.login(loginParams)
+    } catch (e) {
+      Error.sendOnce('store', 'login', 'error logging in', e)
+      throw e
+    }
+
+    // login was successful, use masqStore as abstractStore until logout
+    this.loggedIn = true
+    this.abstractStore = this.masqStore
   }
-}
 
-Store.prototype.setLastLocation = async function(loc) {
-  try {
-    return await abstractStore.set(`qmaps_v${version}_last_location`, loc)
-  } catch (error) {
-    Error.sendOnce('store', 'setLastLocation', 'error setting location', e)
+  async logout() {
+    await this.checkInit()
+    try {
+      await this.masqStore.logout()
+    } catch (e) {
+      Error.sendOnce('store', 'logout', 'error logging out', e)
+      throw e
+    }
+    this.loggedIn = false
+    this.abstractStore = this.localStore
   }
-}
 
-Store.prototype.isRegistered = async function () {
-  try {
-    return await abstractStore.getAllPois() !== null
-  } catch (error) {
-    return false
+  async isLoggedIn() {
+    await this.checkInit()
+    try {
+      return await this.masqStore.isLoggedIn()
+    } catch (e) {
+      Error.sendOnce('store', 'isLoggedIn', 'error checking if logged in', e)
+      throw e
+    }
   }
-}
 
-Store.prototype.onConnect = async function () {
-  return await abstractStore.onConnect()
-}
+  async getUserInfo() {
+    await this.checkInit()
+    try {
+      return await this.masqStore.getUserInfo()
+    } catch (e) {
+      Error.sendOnce('store', 'getUserInfo', 'error getting user info', e)
+      throw e
+    }
+  }
 
-Store.prototype.register = async function () {
-  let regParams = {
-    endpoint: moduleConfig.endpoint,
-    url: window.location.origin + window.location.pathname,
-    title: moduleConfig.masq.title,
-    desc: moduleConfig.masq.desc,
-    icon: moduleConfig.masq.icon
+  async getAllPois() {
+    await this.checkInit()
+    try {
+      return await this.abstractStore.getAllPois()
+    } catch (e) {
+      Error.sendOnce('store', 'getAllPois', 'error getting pois', e)
+      throw e
```

Comment:
this file know wich store is used, so i prefer a more explicit error with the current store used, plus it's not necessary to re throw the excpetion, you can handle here with return []